core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -47,10 +47,32 @@ public abstract class BaseVFSConnectionProvider<T extends VFSConnectionDetails>
     return (List<T>) connectionManagerSupplier.get().getConnectionDetailsByScheme( getKey() );
   }
 
+  /**
+   * Ensure that roles contain the Administrator role when saving/updating connection instance
+   * <p>
+   * @param connectionDetails The connection details
+   */
   @Override public T prepare( T connectionDetails ) throws KettleException {
+    if ( !connectionDetails.getBaRoles().contains( "Administrator" ) ) {
+      connectionDetails.getBaRoles().add( "Administrator" );
+    }
     return connectionDetails;
   }
 
+  /**
+   * Ensure the default roles for new connections or those saved to the meta-store via previous versions of the code.
+   * <p>
+   * @param connectionDetails The connection details
+   */
+  @Override
+  public void hydrate( VFSConnectionDetails connectionDetails ) {
+    List<String> baRoles = connectionDetails.getBaRoles();
+    if ( baRoles.isEmpty() ) {
+      baRoles.add( "Administrator" );
+      baRoles.add( "Authenticated" );
+    }
+  }
+
```

Comment:
This seems like you're putting unrelated business logic in the storage layer. Feels like this would be better somewhere in the auth layers. 
That would also allow the defaults to be changed dynamically, rather than be hard-coded.